scripts/zpool

```diff
@@ -53,7 +53,22 @@ zpool_monitor() {
     # exit immediately if configuration is not valid
     zpool_validate_all || exit $?
 
-    ocf_run -q -info zpool status $OCF_RESKEY_pool
+    if [ -z "$OCF_RESKEY_pool" ] ; then
+        # We have no pool name
+        false
+    elif [ -e /proc/spl/kstat/zfs/$OCF_RESKEY_pool/state ] ; then
+        # Best case, we can read the pool health locklessly from the proc file.
+        state=$(cat /proc/spl/kstat/zfs/$OCF_RESKEY_pool/state)
+        [ "$state" != "SUSPENDED" ] && [ "$state" != "FAULTED" ] && [ "$state" != "UNAVAIL" ]
+    elif [ -d /proc/spl/kstat/zfs/$OCF_RESKEY_pool ] ; then
+        # 2nd best case, we see a /proc entry for the pool so
+        # we at least know it's imported.
+        true
+    else
+        # Worst case, run zpool status to check if the pool is imported.  This
+        # is the worst case since 'zpool status' can hang.
+        ocf_run -q -info zpool status $OCF_RESKEY_pool
+    fi
 
     # This example assumes the following exit code convention
```

Comment:
This comment is now wrong, but if you move to setting rc explicitly under each conditional (and logging a unique message), then that will address this.